language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/util/documentsymbol/DocumentSymbolResolver.java

```diff
@@ -151,20 +182,184 @@
+        name.append(" ").append(serviceDeclarationNode.absoluteResourcePath().stream()
+                .map(Node::toSourceCode).collect(Collectors.joining("")));
+        SymbolKind symbolKind = SymbolKind.Object;
         Range range = DocumentSymbolUtil.generateNodeRange(serviceDeclarationNode);
         Optional<MetadataNode> metadata = serviceDeclarationNode.metadata();
         boolean isDeprecated = metadata.isPresent() &&
                 DocumentSymbolUtil.isDeprecated(metadata.get());
         List<DocumentSymbol> children = transformMembers(serviceDeclarationNode.members());
+        return Optional.ofNullable(createDocumentSymbol(name.toString(), symbolKind, null,
+                range, range, isDeprecated, children, this.context));
+    }
+
+    @Override
+    public Optional<DocumentSymbol> transform(TypeDefinitionNode typeDefinitionNode) {
+        String name = typeDefinitionNode.typeName().text();
+        Node typeDescriptor = typeDefinitionNode.typeDescriptor();
+        SymbolKind symbolKind;
+        List<DocumentSymbol> children = new ArrayList<>();
+        switch (typeDescriptor.kind()) {
+            case RECORD_TYPE_DESC:
+                symbolKind = SymbolKind.Struct;
+                RecordTypeDescriptorNode recordTypeDescriptorNode = (RecordTypeDescriptorNode) typeDescriptor;
+                children.addAll(transformMembers(recordTypeDescriptorNode.fields()));
+                Optional<RecordRestDescriptorNode> restTypeDec = recordTypeDescriptorNode.recordRestDescriptor();
+                if (restTypeDec.isPresent()) {
+                    Optional<DocumentSymbol> restDocSymbol = restTypeDec.get().apply(this);
+                    restDocSymbol.ifPresent(children::add);
+                }
+                break;
+            case OBJECT_TYPE_DESC:
+                symbolKind = SymbolKind.Interface;
+                children.addAll(transformMembers(((ObjectTypeDescriptorNode) typeDescriptor).members()));
+                break;
+            default:
+                symbolKind = SymbolKind.TypeParameter;
+        }
+        Range range = DocumentSymbolUtil.generateNodeRange(typeDefinitionNode);
+        Optional<MetadataNode> metadata = typeDefinitionNode.metadata();
+        boolean isDeprecated = metadata.isPresent() &&
+                DocumentSymbolUtil.isDeprecated(metadata.get());
         return Optional.ofNullable(createDocumentSymbol(name, symbolKind, null,
                 range, range, isDeprecated, children, this.context));
     }
 
+    @Override
+    public Optional<DocumentSymbol> transform(ModuleVariableDeclarationNode moduleVariableDeclarationNode) {
+        BindingPatternNode bindingPatternNode = moduleVariableDeclarationNode.typedBindingPattern().bindingPattern();
+        //Only capture binding pattern is considered. Wildcard and other binding patterns are ignored.
+        String name = bindingPatternNode.kind() == SyntaxKind.CAPTURE_BINDING_PATTERN ?
```

Comment:
When the name is empty, we should not proceed to the symbol generation

Reply:
Done